public/diagnostic.js

```diff
@@ -23,3 +23,34 @@
 //  -  Order 1 : 2 hats ($5 each)
 //  -  Order 2 : 1 desk lamp ($20 each)
 //  -  Order 3 : 3 hand towels ($8 each)
+
+const Order = Ember.Object.extend({
+  unitPrice: NaN,
+  quantity: NaN,
+  name: '',
+  orderPrice: Ember.computed('unitPrice', 'quantity', function() {
+    return unitPrice * quantity;
+  }),
+});
+
+const Cart = Ember.Object.extend({
+  orders: [],
+  addToCart: function(order) {
+    this.orders.push(order);
+  },
+  totalPrice: Ember.computed('orders.@each', function() {
+    let hold = orders.map((val) => { return val.orderPrice; });
+      return hold.reduce((a,b) => a + b);
+  }),
+});
+
+let order1 = Order.create({ unitPrice: 5, quantity = 2, name: "hat" });
+let order2 = Order.create({ unitPrice: 20, quantity = 1, name: "desk lamp" });
+let order3 = Order.create({ unitPrice: 8, quantity = 3, name: "hand towel" });
+
+let cart = Cart.create();
+cart.addToCart(order1);
+cart.addToCart(order2);
+cart.addToCart(order3);
+
+cart.totalPrice;
```

Comment:
your cart and order instances look good.  Though, instead of `cart.totalPrice` use `console.log(cart.get('totalPrice'))`